src/db/mysql/MySqlAPI.cpp

```diff
@@ -566,6 +566,69 @@ static int getActiveCount(soci::session& sql, const std::string &source, const s
     return activeCount;
 }
 
+/// Determines if the link has a defined upper bound in the 
+// database and returns the number of files allowed for transfer on the specified link
+/// @param source Source storage
+/// @param dest Destination storage
+/// @return bool indicating if there is an upper bound capacity on link
+/// @return int representing maxActive on link
+/// @return int representing currentActive on link
+boost::tuple<bool, int> linkUpperBound(soci::session& sql, std::string sourceSe, std::string destSe) {
+    int maxActive = 0;
+    soci::indicator maxActiveNull = soci::i_ok;
+    int activeCount = getActiveCount(sql, sourceSe, destSe);
+    int filesNum = 10;
+
+    sql << "SELECT active FROM t_optimizer WHERE source_se = :source_se AND dest_se = :dest_se",
+        soci::use(sourceSe),
+        soci::use(destSe),
+        soci::into(maxActive, maxActiveNull);
+
+    if (maxActiveNull != soci::i_null && maxActive > 0) {
+        filesNum = maxActive - activeCount;
+        return boost::make_tuple(true, filesNum);
+    }
+    return boost::make_tuple(false, filesNum);
+}
+
+std::map<Pair, int> MySqlAPI::getLinkCapacities(const std::vector<QueueId>& queues,
+        std::map<std::string, std::list<TransferFile> >& files)
+{
+    std::map<Pair, int> linkCapacities;
+    soci::session sql(*connectionPool);
+    time_t now = time(NULL);
+    try
+    {
+        // Iterate through queues, setting link capacities as VO credits
+        for (auto it = queues.begin(); it != queues.end(); ++it)
+        {
+            boost::tuple<bool, int> linkInfo = linkUpperBound(sql, it->sourceSe, it->destSe);
```

Comment:
Introducing new db call? - impact on runtime

Reply:
I'm not sure what you mean about additional db call?